module-core/src/main/java/xyz/voltawallet/internal/VoltaFactory.java

```diff
@@ -34,4 +46,27 @@ public static VoltaClient createVoltaClient(String bundleServiceUrl) {
     }
     return instance;
   }
+
+  public static VaultClient createVaultClient(long chainId, String bundleServiceUrl, ContractAddresses contractAddresses) {
+    Web3j web3j = Web3j.build(new HttpService(bundleServiceUrl));
+    TransactionManager transactionManager = new ReadonlyTransactionManager(web3j, contractAddresses.getEntryPoint());
+    ContractGasProvider contractGasProvider = new StaticEIP1559GasProvider(
+      chainId,
+      new BigInteger("39000000000"),
+      new BigInteger("218000000000"),
+      new BigInteger("21000")
+    );
```

Comment:
Just curious, is it safe to use hardcode gas when initializing VaultClient?

Reply:
The sdk doesn't use the values for current features, the gas for User Operation will be estimated when building the User Operation.  But I will search a solution to provide correct gas value for Volta contract components.